### ISSUE TITLE
Content update fails with `Cannot access 'k' before initialization`

### ISSUE THREAD

**Issue by tbe** [vugu/vugu#117]
**Describe the bug**
During a transition to a different component on a side, the update fails sometimes with `Cannot access 'k' before initialization`.

The behavior depends on the formatting of the component.

**Software Versions**
Vugu version: 0.2.0
Go version: 1.14.0
Browser and version: Google Chrome 80.0.3987.87 (Linux)

**To Reproduce**
_Failing_
products.vugu:
```
<section class="section">
    <div class="container">
        <h1 class="title">Managed Products</h1>
        <div id="content">
            <a class="button is-pulled-right" @click='c.Navigate("/products/create",nil, vgrouter.NavReplace)'>Create
                new product</a>
            <table class="table" style="width: 100%">
                <!-- content here -->
            </table>
        </div>
    </div>
</section>
```
create.vugu:
```
<section class="section">
    <div class="container">
        <h1 class="title">Create a new product</h1>
        <div id="content">
            <form id="myform">
              <!-- content here -->
            </form>
        </div>
    </div>
</section>
```
_working_:
Same products.vugu as above, but lightly changed create.vugu:
```
<section class="section">
    <div class="container">
        <h1 class="title">Create a new product</h1>
        <div id="content"><form id="myform">
              <!-- content here -->
            </form></div>
    </div>
</section>
```

**Expected behavior**
Transition works always.

**Additional Notes**
Full error message from WASM:
![image](https://user-images.githubusercontent.com/102155/78799091-0439db80-79ba-11ea-9e61-4daee9745968.png)

**Comment by bradleypeabody** [comment]
Thanks for the information on this.  I had look at this but so far have been unable to reproduce it.

If this is worth looking into further, we can do the following:

I made a branch https://github.com/vugu/vugu/tree/issue-117 for this and put a test case at `wasm-test-suite/test-013-issue-117/root.vugu`.  To run it you'll just need to run `run-wasm-test-suite-docker.sh` in the `wasm-test-suite` folder - this will start a docker container that the test suite uses.  Then you can run `Test013Issue117` (e.g. `go test . -v -run Test013Issue117` from that same folder).  It will try to build and load the page.  It will also output the URL which you can browse to manually and look at the console.  If the problem can show up in this test case then I should be able to track it down.

**Comment by bradleypeabody** [comment]
Thanks for your work on this.

Turns out it was something really simple (https://github.com/vugu/vugu/commit/51d3713e37e784519a4b001cc93e7e0905bc3396#diff-b9df313abb8f7b24742a2b37c1a54971) with a var being used before being declared (just one more reason I like writing Go code instead of JS ;) )

The only mystery is how the bug wasn't being triggered by any of the other tests.  In any case, all fixed now and released as v0.2.1
